src/toil/lib/aws/__init__.py

```diff
@@ -11,22 +11,258 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Caches all boto3 clients we instantiate."""
+import collections
+import inspect
+import logging
+import os
 import re
+import socket
+import threading
 from functools import lru_cache
+from urllib.request import urlopen
+from urllib.error import URLError
 
+from typing import Any, Callable, Dict, Iterable, List, Optional, TypeVar, Union
 
-@lru_cache(maxsize=None)
-def client(*args, **kwargs):
-    import boto3
-    return boto3.client(*args, **kwargs)
+logger = logging.getLogger(__name__)
 
+try:
+    # This file contains some stuff that needs boto/boto3 to be installed in
+    # order to be defined. But this __init__ file and files under it also
+    # contain some stuff that we need to be able to import when boto isn't
+    # installed. So everything that does need those modules gets defined in
+    # this try.
```

Comment:
I guess I'm voting that this be put into a non-init file to avoid giant try/excepts like this.

Reply:
I'll move these to another file.